### PR TITLE
Azure SSI SDK

### PR DESCRIPTION
**Pull request by chandra1-n**
Java SDK for import vm, import status, import configuration, node logs, node resources

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeDiskModeEnum.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.spotinst.sdkjava.enums;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public enum AzureStatefulNodeDiskModeEnum {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeDiskModeEnum.java:7]: Move this to a folder (azure.statefulNode) inside the enums folder.

Comment (anuragsharma-123) [reply src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeDiskModeEnum.java:7]: Rename to AzureStatefulNodeDiskPersistenceModeEnum

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeLifeCycleTypeEnum.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package com.spotinst.sdkjava.enums;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public enum AzureStatefulNodeLifeCycleTypeEnum {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeLifeCycleTypeEnum.java:7]: Rename to AzureStatefulNodePreferredLifeCycleEnum

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeLifeCycleTypeEnum.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    //endregion
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStatefulNodeLifeCycleTypeEnum.class);
+
+    //region methods
+    public static AzureStatefulNodeLifeCycleTypeEnum fromName(String name) {
+        AzureStatefulNodeLifeCycleTypeEnum retVal = null;
+
+        for (AzureStatefulNodeLifeCycleTypeEnum productEnum : AzureStatefulNodeLifeCycleTypeEnum.values()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeLifeCycleTypeEnum.java:29]: Change iterator "productEnum" to "enumValue"

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeOsEnum.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    //endregion
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStatefulNodeOsEnum.class);
+
+    //region methods
+    public static AzureStatefulNodeOsEnum fromName(String name) {
+        AzureStatefulNodeOsEnum retVal = null;
+
+        for (AzureStatefulNodeOsEnum productEnum : AzureStatefulNodeOsEnum.values()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeOsEnum.java:29]: Change iterator "productEnum" to "enumValue"

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodePerformAtEnum.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    private              String name;
+
+    private AzureStatefulNodePerformAtEnum(String name) {
+        this.name = name;
+    }
+
+    public static AzureStatefulNodePerformAtEnum fromName(String name) {
+        AzureStatefulNodePerformAtEnum retVal = null;
+
+        for (AzureStatefulNodePerformAtEnum instanceMarketEnum : AzureStatefulNodePerformAtEnum.values()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodePerformAtEnum.java:22]: Change iterator "instanceMarketEnum" to "enumValue"

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeStateEnum.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package com.spotinst.sdkjava.enums;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public enum AzureStatefulNodeStateEnum {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeStateEnum.java:6]: Should it be renamed to AzureStatefulNodeSchedulingTaskTypeEnum

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeStateEnum.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    private              String name;
+
+    private AzureStatefulNodeStateEnum(String name) {
+        this.name = name;
+    }
+
+    public static AzureStatefulNodeStateEnum fromName(String name) {
+        AzureStatefulNodeStateEnum retVal = null;
+
+        for (AzureStatefulNodeStateEnum instanceMarketEnum : AzureStatefulNodeStateEnum.values()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeStateEnum.java:22]: Change iterator to "enumValue"

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeTaskTypeEnum.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
+package com.spotinst.sdkjava.enums;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public enum AzureStatefulNodeTaskTypeEnum {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeTaskTypeEnum.java:7]: Should it be renamed to AzureStatefulNodeSchedulingTaskTypeEnum

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeTaskTypeEnum.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    //endregion
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStatefulNodeTaskTypeEnum.class);
+
+    //region methods
+    public static AzureStatefulNodeTaskTypeEnum fromName(String name) {
+        AzureStatefulNodeTaskTypeEnum retVal = null;
+
+        for (AzureStatefulNodeTaskTypeEnum productEnum : AzureStatefulNodeTaskTypeEnum.values()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeTaskTypeEnum.java:30]: rename iterator to enumValue

File: src/main/java/com/spotinst/sdkjava/example/statefulnode/azure/AzureStatefulNodeExample.java [hunk 10]
```diff
@@ -0,0 +1,849 @@
...
+
+        List<LaunchSpecExtensionsSpecification> extensionsList = Collections.singletonList(extensions);
+        // Build Login
+        LaunchSpecLoginSpecification login =
+                LaunchSpecLoginSpecification.Builder.get().setUserName("ubuntu").setPassword("NetApp@123@321")
+//                        .setSshPublicKey("ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC6v8BnN6OcFZjDLQ85uSg3qM/p2WVddk8J2S921uO8ydT1M3lwy+vSNWT6O7/wUl2U0c+ZcFJSEZQCLJ7cs85Q9ER6b9oscABLFtnYdTR9OBNJ9B9oTkao+EgEEa3i8uX2iMzqVZndQJoJ1/N3ds1KhozKC2t76jD+rPRjHQJ4ReJHNHO+aalivssPwfofELg82dJ1urWksjXSdzO39OHBqfCIztS1wPeiWWYSuJWJuPL000bfH8ngU5Vzh0plPK9fdRmBIEx8GhY4hBfOSlRO5ITaIqQTXoZaMHCX2AwhIj+ZHiiWPY+5/9x9H6tdLXRJ9huCF5dNaTj2D8Jt1So1B6QuN8Iqchu7FzlpuSB+uOaChvJ5NfGEJvCO7SqosiSKhxOv0GAFY99Vj53JoUO3+7mFortO+kDmMKwrJmw0adTURHM+tetNd6txs+86FmU576b3MhvTBbssCH1A54gThdbtseOEqrRJMNQoicb0f2/IzkdjT6RWu4IG+vFMbHLOts1dDqP3paWY/vhHfTvNVcXU5gYzu4RZZOtespRt3/kSBgiZvmhiifVqShf6cgn6+9BGznT4FMtpQZQ9tqP/hUII/9uQn7CEU6X7Pualc7FiWjGbEVArTVHHTxIfUPTqnp9f7X1oG4+AYvSTQsJXJalrSBx6iok4+9Xk5pxrVQ==")
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/example/statefulnode/azure/AzureStatefulNodeExample.java:188]: remove this commented statement

File: src/main/java/com/spotinst/sdkjava/example/statefulnode/azure/AzureStatefulNodeExample.java [hunk 11]
```diff
@@ -0,0 +1,849 @@
...
+        //Build Extentions
+        LaunchSpecExtensionsSpecification.Builder extensionBuilder = LaunchSpecExtensionsSpecification.Builder.get();
+        LaunchSpecExtensionsSpecification extensions = extensionBuilder.setName("Automation-Extension").setType("customScript")
+                                                                       .setPublisher("Microsoft.Azure.Extensions").setApiVersion("2.0")
+                                                                       .setMinorVersionAutoUpgrade(true).setProtectedSettings(protectedSettings).build();
+
+        List<LaunchSpecExtensionsSpecification> extensionsList = Collections.singletonList(extensions);
+        // Build Login
+        LaunchSpecLoginSpecification login =
+                LaunchSpecLoginSpecification.Builder.get().setUserName("ubuntu").setPassword("NetApp@123@321")
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/example/statefulnode/azure/AzureStatefulNodeExample.java:187]: declare username and password as private final string like auth_token and use. We don't want these values to be in the code.

File: src/main/java/com/spotinst/sdkjava/example/statefulnode/azure/AzureStatefulNodeExample.java [hunk 12]
```diff
@@ -0,0 +1,849 @@
...
+        List<HealthCheckTypeEnumAzure> healthCheckTypesList = new ArrayList<>();
+        healthCheckTypesList.add(HealthCheckTypeEnumAzure.VM_STATE);
+        StatefulNodeHealthConfiguration.Builder healthBuilder = StatefulNodeHealthConfiguration.Builder.get();
+        StatefulNodeHealthConfiguration health =
+                healthBuilder.setHealthCheckTypes(healthCheckTypesList).setUnhealthyDuration(300).setGracePeriod(180).setAutoHealing(true).build();
+
+        // Build Stateful Node
+        StatefulNode.Builder statefulNodeBuilder = StatefulNode.Builder.get();
+        StatefulNode statefulNode =
+                statefulNodeBuilder.setName("Automation-java-SDK-StatefulNode").setRegion("eastus").setResourceGroupName("ManualQAResourceGroup")
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/example/statefulnode/azure/AzureStatefulNodeExample.java:259]: Automation should be using resources from AutomationResourceGroup. Please change throughout the examples.

File: src/main/java/com/spotinst/sdkjava/model/ISpotAzureStatefulNodeRepo.java [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+    default RepoGenericResponse<Boolean> deleteNode(StatefulNodeDeallocationConfig statefulNodeDeallocationConfig, String nodeId, String authToken, String account){
+        throw new SpotinstNotSupportedException();
+    }
+
+    default RepoGenericResponse<List<StatefulNode>> getAllNodes(String authToken, String account){
+        throw new SpotinstNotSupportedException();
+    }
+
+    default RepoGenericResponse<Boolean> updateNodeState(StatefulNodeStateChangeRequest recycleStatefulNodeRequest, String nodeId, String authToken, String account){
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ISpotAzureStatefulNodeRepo.java:34]: Change variable name "recycleStatefulNodeRequest" to something generic, not specific to recycle

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecCustomConfiguration.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiLaunchSpecCustomConfiguration implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecCustomConfiguration.java:15]: Where this object is used?

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsProtectedSettings.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+/**
+ * Created by Bansi Chapla on 23/11/2021.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiLaunchSpecExtensionsProtectedSettings implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsProtectedSettings.java:17]: Where did u get the fields for this object?

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsSpecification.java [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiLaunchSpecExtensionsSpecification implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsSpecification.java:15]: publicSettings field is missing (refer - https://docs.spot.io/api/#operation/azureStatefulNodeCreate)

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecGalleryConfiguration.java [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiLaunchSpecGalleryConfiguration implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecGalleryConfiguration.java:15]: Rename to ApiLaunchSpecImageGalleryConfiguration. Same change for BL class also.

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskConfiguration.java [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeAttachDataDiskConfiguration implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String> isSet;
+    private String          dataDiskName;
+    private String          dataDiskResourceGroupName;
+    private Integer         lun;
+    private Integer         sizeGB;
+    private String          storageAccountType;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskConfiguration.java:23]: Should this field be enumeration?

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskResponse.java [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+public class ApiStatefulNodeAttachDataDiskResponse implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                     isSet;
+    private String                          name;
+    private String                          resourceGroupName;
+    private String                          region;
+    private String                          zone;
+    private Integer                         size;
+    private Integer                         lun;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskResponse.java:24]: Should be enumeration.

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeComputeConfiguration.java [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+public class ApiStatefulNodeComputeConfiguration implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                         isSet;
+    private String                              os;
+    private ApiStatefulNodeVmSizesConfiguration vmSizes;
+    private List<String>                        zones;
+    private String                              preferredZone;
+    private ApiStatefulNodeLaunchSpecification  launchSpecification;
+    private ApiStatefulNodeLoadBalancersConfig  loadBalancersConfig;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeComputeConfiguration.java:28]: LoadbalancerConfig is not a field in this object as per documentation.

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeComputeConfiguration.java [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeComputeConfiguration implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                         isSet;
+    private String                              os;
+    private ApiStatefulNodeVmSizesConfiguration vmSizes;
+    private List<String>                        zones;
+    private String                              preferredZone;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeComputeConfiguration.java:26]: preferredZone should be preferredZones and it should be an array of strings.

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeDeallocationConfig.java [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeDeallocationConfig implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                               isSet;
+    private Boolean                                   shouldTerminateVm;
+    private ApiStatefulNodeNetworkDeallocationConfig  networkDeallocationConfig;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeDeallocationConfig.java:23]: Do we need 4 different structures. They have same 2 parameters.

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeHealthConfiguration.java [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeHealthConfiguration implements IPartialUpdateEntity {
+    @JsonIgnore
+    private Set<String>  isSet;
+    private Boolean      autoHealing;
+    private Integer      gracePeriod;
+    private List<String> healthCheckTypes;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeHealthConfiguration.java:21]: should this be an enumeration

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeLaunchSpecification.java [hunk 24]
```diff
@@ -0,0 +1,138 @@
...
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeLaunchSpecification implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeLaunchSpecification.java:16]: BootDiagnostics, Customdata, loadbalancerconfig,managedserviceidentities, secrets, shutdown script parameters missing

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeNetworkDeallocationConfig.java [hunk 25]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by Bansi Chapla on 23/11/2021.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeNetworkDeallocationConfig implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeNetworkDeallocationConfig.java:18]: Same configuration as diskconfiuration can be used

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodePublicIpDeallocationConfig.java [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by Bansi Chapla on 23/11/2021.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodePublicIpDeallocationConfig implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodePublicIpDeallocationConfig.java:18]: redundant. Use same as disk configuration. Applies to BL classes also.

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkInterfaces.java [hunk 27]
```diff
@@ -0,0 +1,149 @@
...
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeResourceNetworkInterfaces implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                                         isSet;
+    private String                                              name;
+    private String                                              resourceGroupName;
+    private Boolean                                             isPrimary;
+    private ApiLaunchSpecNetworkSecurityGroupConfiguration      networkSecurityGroup;
+    private List<ApiStatefulNodeResourceNetworkIpConfigurations>networkIpConfigurations;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkInterfaces.java:25]: no space b/w variable name and variable type.
variable name should be changed to ipConfigurations.

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkIpConfigurations.java [hunk 28]
```diff
@@ -0,0 +1,151 @@
...
+public class ApiStatefulNodeResourceNetworkIpConfigurations implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                             isSet;
+    private String                                  name;
+    private String                                  privateIpAddress;
+    private String                                  privateIPAddressVersion;
+    private String                                  privateIpAllocationMethod;
+    private String                                  applicationSecurityGroups;
+    private Boolean                                 primary;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkIpConfigurations.java:24]: variable name should be isPrimary

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceOsDisk.java [hunk 29]
```diff
@@ -0,0 +1,132 @@
...
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeResourceOsDisk implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>             isSet;
+    private String                  name;
+    private String                  resourceGroupName;
+    private String                  osType;
+    private String                  storageAccountType;
+    private Integer                 sizeGb;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceOsDisk.java:23]: sizeGB

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeSnapshotDeallocationConfig.java [hunk 30]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by Bansi Chapla on 23/11/2021.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiStatefulNodeSnapshotDeallocationConfig implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeSnapshotDeallocationConfig.java:18]: Reuse existing class. Redundant.

File: src/main/java/com/spotinst/sdkjava/model/bl/azure/statefulNode/LaunchSpecCustomConfiguration.java [hunk 31]
```diff
@@ -0,0 +1,89 @@
+package com.spotinst.sdkjava.model.bl.azure.statefulNode;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.HashSet;
+import java.util.Set;
+
+public class LaunchSpecCustomConfiguration {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/azure/statefulNode/LaunchSpecCustomConfiguration.java:7]: couldn't find where this object is used.

**Review by anuragsharma-123 — COMMENTED** [review]
Comments given for Api classes are applicable for BL classes also.

File: src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeDiskModeEnum.java (hunk 1 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/enums/AzureStatefulNodeDiskModeEnum.java:7]: resolved

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiImportConfiguration.java [hunk 32]
```diff
@@ -0,0 +1,121 @@
...
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiImportConfiguration implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                          isSet;
+    private String                               originalVmName;
+    private Integer                              drainingTimeout;
+    private Integer                              resourcesRetentionTime;
+    private String                               resourceGroupName;
+    @JsonProperty("node")
```
[the remark this answers is not in the document]
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiImportConfiguration.java:20]: import body will take node as object inside statefulNodeImport object, for that we need to specify the node

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecCustomConfiguration.java (hunk 14 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecCustomConfiguration.java:15]: this is used, when we used custom image

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsProtectedSettings.java (hunk 15 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsProtectedSettings.java:17]: used in extensions

Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsProtectedSettings.java:17]: from postman collection shared by dev team

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsSpecification.java (hunk 16 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecExtensionsSpecification.java:15]: on hold for clarification, will fix as part of document review

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecNetworkInterfacesConfiguration.java [hunk 33]
```diff
@@ -0,0 +1,110 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiLaunchSpecNetworkInterfacesConfiguration implements IPartialUpdateEntity {
```
[the remark this answers is not in the document]
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecNetworkInterfacesConfiguration.java:15]: network block fix is needed in documentation

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecOsDiskSpecification.java [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiLaunchSpecOsDiskSpecification implements IPartialUpdateEntity {
+    @JsonIgnore
+    private Set<String> isSet;
+    private String      name;
+    private Integer     sizeGB;
+    private String      type;
```
[the remark this answers is not in the document]
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiLaunchSpecOsDiskSpecification.java:20]: new enum file added

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskConfiguration.java (hunk 18 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskConfiguration.java:23]: common enum "AzureStatefulNodeDiskTypeEnum" file added for both os and data disk types

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskResponse.java (hunk 19 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeAttachDataDiskResponse.java:24]: lun is not enum

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeComputeConfiguration.java (hunk 21 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeComputeConfiguration.java:26]: document error

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeDeallocationConfig.java (hunk 22 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeDeallocationConfig.java:23]: I will make a note of this and make changes if needed

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeHealthConfiguration.java (hunk 23 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeHealthConfiguration.java:21]: changed

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeLaunchSpecification.java (hunk 24 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeLaunchSpecification.java:16]: Need clarification on this

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkInterfaces.java [hunk 35]
```diff
@@ -0,0 +1,149 @@
...
+    private String                                              macAddress;
```
[the remark this answers is not in the document]
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkInterfaces.java:26]: noted for clarification

File: src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkIpConfigurations.java (hunk 28 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/api/azure/statefulNode/ApiStatefulNodeResourceNetworkIpConfigurations.java:24]: clarification

File: src/main/java/com/spotinst/sdkjava/model/bl/azure/statefulNode/LaunchSpecCustomConfiguration.java (hunk 31 above)
Comment (chandra1-n) [reply src/main/java/com/spotinst/sdkjava/model/bl/azure/statefulNode/LaunchSpecCustomConfiguration.java:7]: this is used, when we use custom image

**Comment by anuragsharma-123** [comment]
Outdated PR. Azure SSI SDK implementation is merged through another PR. Closing this PR without merging.

**Pull request closed without merging**